### DISCUSSION TITLE
Group mandatory modules in one file and more

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2570] [Ideas]
Let's not lose the ideas presented in this PR #2455.

**We have 3 types of modules**:

- level 1: mandatory for OpenMage to work and on which the modules in levels 2 and 3 depend.
- level 2: which do not depend on modules from level 3
- level 3: which depend on the modules from level 2

1. Grouping of mandatory modules in one file. The file name can be Mage_Require.xml. In this way, the non-functionality of the store is prevented due to an editing error. The file will contain an explanatory warning comment.

2. Explanatory warning comments in the level 2 modules to prevent non-functionality of the dependencies. Also, the list of modules dependent on it will be specified in each file in order to evaluate immediately what is the impact on functionality.

3. Establishing the moment when the Mage_All.xml file will be removed from the repository or if it will be kept permanently.

4. Analyzing if there is a better method to control the order in which the modules are loaded, if the current order is the optimal one.

**Comment by kiatng** [comment]
To remove a module from levels 2 and 3, we need to remove the files in these folders:

1. app\etc\modules\module.xml
2. app\code\core\Mage\module all files
3. app\code\core\Mage\Adminhtml\ (not easy, but PR#2533 will fix this)
4. app\design\adminhtml\default\default\layout\module.xml
5. app\design\adminhtml\default\default\template\module all files
6. app\design\adminhtml\default\openmage\layout\module.xml
5. app\design\adminhtml\default\openmage\template\module all files
6. app\design\frontend\base\default\layout\module.xml
7. app\design\frontend\base\default\template\module all files
8. app\design\frontend\rwd\default\layout\module.xml
7. app\design\frontend\rwd\default\template\module all files
8. May be other files in skin folder, js folder need to be removed
9. Tables in the database may need to be removed
10. May be associated rows in table `core_config_data` and `core_resource` may need to be removed

I probably miss other stuff. A script would be nice.

**Reply by addison74** [reply]
My post does not intend to remove any module, but only to clarify aspects of functionality of the modules when the administrator makes the decision to edit these files. They are those aspects that were discussed in the PR but it was decided to be implemented later.
